src/utils/utils.cpp

```diff
@@ -345,6 +348,191 @@ void PrintValue(std::ostream &os, const mg_value *value) {
+}
+
+ReplxxColor GetWordColor(const std::string_view word) {
+  auto word_uppercase = utils::ToUpperCase(std::string(word));
+  bool is_cypher_keyword =
+      std::find(constants::kCypherKeywords.begin(),
+                constants::kCypherKeywords.end(),
+                word_uppercase) != constants::kCypherKeywords.end();
+  bool is_memgraph_keyword =
+      std::find(constants::kMemgraphKeywords.begin(),
+                constants::kMemgraphKeywords.end(),
+                word_uppercase) != constants::kMemgraphKeywords.end();
+  bool is_awesome_function =
+      std::find(constants::kAwesomeFunctions.begin(),
+                constants::kAwesomeFunctions.end(),
+                word_uppercase) != constants::kAwesomeFunctions.end();
+  if (is_cypher_keyword || is_memgraph_keyword) {
+    return REPLXX_COLOR_YELLOW;
+  } else if (is_awesome_function) {
+    return REPLXX_COLOR_BRIGHTRED;
+  } else {
+    return REPLXX_COLOR_DEFAULT;
+  }
+}
+
+void SetWordColor(std::string_view word, ReplxxColor *colors,
+                  int *colors_offset) {
+  auto color = GetWordColor(word);
+  auto word_codepoint_len = utf8str_codepoint_length(word.data(), word.size());
+  for (int i = *colors_offset; i < *colors_offset + word_codepoint_len; i++) {
+    colors[i] = color;
+  }
+  // +1 for the word boundary char (we don't want to color it)
+  *colors_offset = *colors_offset + word_codepoint_len + 1;
+}
+
+void ColorHook(const char *input, ReplxxColor *colors, int size, void *) {
+  auto *word_begin = input;
+  size_t word_size = 0;
+  int colors_offset = 0;
+
+  auto input_size = strlen(input);
+  for (int i = 0; i < input_size; i++) {
+    // word boundary
+    if (strchr(wb, input[i]) != NULL) {
+      SetWordColor(std::string_view(word_begin, word_size), colors,
+                   &colors_offset);
+      // if the boundary is not the last char in input, advance the
+      // next word ptr and reset word size
+      if (i != (input_size - 1)) {
+        word_begin = input + i + 1;
+        word_size = 0;
+      }
+      // end of input
+    } else if (i == (input_size - 1)) {
+      // regular char encountered as the last char of input
+      word_size++;
+      SetWordColor(std::string_view(word_begin, word_size), colors,
+                   &colors_offset);
+    } else {
+      // regular char encountered, but not the end of input - advance
+      word_size++;
+    }
+  }
+}
+
+void ParseStats(const mg_value *mg_stats,
+                std::optional<std::map<std::string, std::int64_t>> &ret_stats) {
+  const mg_map *stats = mg_value_map(mg_stats);
+  ret_stats.emplace();
+  for (size_t j = 0; j < mg_map_size(stats); ++j) {
+    const mg_string *mg_stat_key = mg_map_key_at(stats, j);
+    const auto stat_key =
+        std::string(mg_string_data(mg_stat_key), mg_string_size(mg_stat_key));
+
+    const int64_t stat_value = mg_value_integer(mg_map_value_at(stats, j));
+    ret_stats->insert({stat_key, stat_value});
```

Comment:
```suggestion
    auto stat_key =
        std::string(mg_string_data(mg_stat_key), mg_string_size(mg_stat_key));

    const int64_t stat_value = mg_value_integer(mg_map_value_at(stats, j));
    ret_stats->emplace(std::move(stat_key), stat_value);
```

I think this form is better, because:
1. it is more efficient
2. it is simpler because it is lacking the extra `{}`